scripts/ASTImportTest.sh

```diff
@@ -1,105 +1,190 @@
+# (at your option) any later version.
+#
+# solidity is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with solidity.  If not, see <http://www.gnu.org/licenses/>
+#
+# (c) solidity contributors.
+# ------------------------------------------------------------------------------
+# Bash script to test the import/exports.
+# ast import/export tests:
+#   - first exporting a .sol file to JSON, then loading it into the compiler
+#     and exporting it again. The second JSON should be identical to the first.
+
+set -euo pipefail
 
-set -e
-
-# Bash script to test the ast-import option of the compiler by
-# first exporting a .sol file to JSON, then loading it into the compiler
-# and exporting it again. The second JSON should be identical to the first
 READLINK=readlink
 if [[ "$OSTYPE" == "darwin"* ]]; then
     READLINK=greadlink
 fi
 REPO_ROOT=$(${READLINK} -f "$(dirname "$0")"/..)
 SOLIDITY_BUILD_DIR=${SOLIDITY_BUILD_DIR:-${REPO_ROOT}/build}
-SOLC=${SOLIDITY_BUILD_DIR}/solc/solc
-SPLITSOURCES=${REPO_ROOT}/scripts/splitSources.py
+SOLC="${SOLIDITY_BUILD_DIR}/solc/solc"
+SPLITSOURCES="${REPO_ROOT}/scripts/splitSources.py"
+
+# shellcheck source=scripts/common.sh
+source "${REPO_ROOT}/scripts/common.sh"
+
+function print_usage
+{
+    fail "Usage: ${0} ast [--exit-on-error]."
+}
+
+function print_used_commands
+{
+    local test_directory="$1"
+    local export_command="$2"
+    local import_command="$3"
+    printError "You can find the files used for this test here: ${test_directory}"
+    printError "Used commands for test:"
+    printError "# export"
+    echo "$ ${export_command}" >&2
+    printError "# import"
+    echo "$ ${import_command}" >&2
+}
+
+function print_stderr_stdout
+{
+    local error_message="$1"
+    local stderr_file="$2"
+    local stdout_file="$3"
+    printError "$error_message"
+    printError ""
+    printError "stderr:"
+    cat "$stderr_file" >&2
+    printError ""
+    printError "stdout:"
+    cat "$stdout_file" >&2
+}
+
+IMPORT_TEST_TYPE=
+EXIT_ON_ERROR=0
+for PARAM in "$@"
+do
+    case "$PARAM" in
+        ast) IMPORT_TEST_TYPE="ast" ;;
+        --exit-on-error) EXIT_ON_ERROR=1 ;;
+        *) print_usage ;;
+    esac
+done
 
 SYNTAXTESTS_DIR="${REPO_ROOT}/test/libsolidity/syntaxTests"
 ASTJSONTESTS_DIR="${REPO_ROOT}/test/libsolidity/ASTJSON"
-NSOURCES="$(find "$SYNTAXTESTS_DIR" -type f | wc -l)"
-
-# DEV_DIR="${REPO_ROOT}/../tmp/contracts/"
-# NSOURCES="$(find $DEV_DIR -type f | wc -l)" #TODO use find command
 
 FAILED=0
 UNCOMPILABLE=0
 TESTED=0
 
-if [[ "$(find . -maxdepth 0 -type d -empty)" == "" ]]; then
-    echo "Test directory not empty. Skipping!"
-    exit 1
-fi
+function ast_import_export_equivalence
```

Comment:
For consistency with the naming of the other function:
```suggestion
function test_ast_import_export_equivalence
```